### PR TITLE
chore: PlaywrightClient/Server enhancements

### PR DESCRIPTION
**Pull request by mxschmitt**
- ensure timeout results in a meaningful message
- add onDisconnect handler

### CODE REVIEW

File: src/remote/playwrightServer.ts [hunk 1]
```diff
@@ -33,14 +33,15 @@ export interface PlaywrightServerDelegate {
 
 export type PlaywrightServerOptions = {
   acceptForwardedPorts?: boolean
+  onDisconnect?: () => void;
```
Comment (yury-s) [inline src/remote/playwrightServer.ts:36]: I don't see any usages, drop it for now?

Comment (mxschmitt) [reply src/remote/playwrightServer.ts:36]: I use it for a demo, see later in the standup. Since it's only an internal API, it should be fine?

Comment (yury-s) [reply src/remote/playwrightServer.ts:36]: If it's just for the demo you can use a local change? We normally stick to the rule of not adding unused code but if you really need it to be in the repo for the demo it's okay.

**Review by yury-s — APPROVED** [review]

File: src/remote/playwrightServer.ts (hunk 1 above)
Comment (mxschmitt) [reply src/remote/playwrightServer.ts:36]: (not exposed API, thats why I landed it for now. If we publish it at some point we re-consider the whole use of the API)

**Pull request merged**